samza-core/src/main/java/org/apache/samza/clustermanager/AbstractContainerAllocator.java

```diff
@@ -171,7 +174,7 @@ public void requestResources(Map<String, String> resourceToHostMappings) {
     for (Map.Entry<String, String> entry : resourceToHostMappings.entrySet()) {
       String containerId = entry.getKey();
       String preferredHost = entry.getValue();
-      if (preferredHost == null)
+      if (preferredHost == null || !clusterManagerConfig.getHostAffinityEnabled())
```

Comment:
You can make AbstractContainerAllocator.java ignore all present resourceToHostMappings, since AbstractContainerAllocator is instantiated only in case host-affinity is disabled, otherwise we instantiate 
HostAwareContainerAllocator.